framework/CMakeLists.txt

```
@@ -402,6 +402,21 @@ source_group("scene_graph\\components\\" FILES ${SCENE_GRAPH_COMPONENT_FILES})
 source_group("scene_graph\\scripts\\" FILES ${SCENE_GRAPH_SCRIPTS_FILES})
 source_group("stats\\" FILES ${STATS_FILES})
 
+# Adding scene graph files to a seperate library. Libtool doens't support having multiple files sharing the same name.
+set(PROJECT_SCENE_GRAPH_FILES
+    ${SCENE_GRAPH_FILES}
+    ${SCENE_GRAPH_COMPONENT_FILES}
+    ${SCENE_GRAPH_SCRIPTS_FILES}
+)
+
+add_library(framework_scene_graph ${PROJECT_SCENE_GRAPH_FILES})
```

Do not rely on the behavior of `add_library` to default to static.  Be explicit:

```suggestion
add_library(framework_scene_graph STATIC ${PROJECT_SCENE_GRAPH_FILES})
```

If someone has `build_shared_libs` turned on when doing the CMake configure, this will fail to build, because the `framework_scene_graph` and `framework` have bidirectional dependencies, so building the former as a shared library will fail at link time.